com.unity.render-pipelines.universal/Runtime/2D/Passes/Utility/RendererLighting.cs

```diff
@@ -162,32 +163,72 @@ private static void RenderLightSet(IRenderPass2D pass, RenderingData renderingDa
 
                     ShadowRendering.RenderShadows(pass, renderingData, cmd, layerToRender, light, light.shadowIntensity, renderTexture, renderTexture);
 
+                    // Still used by non-Batched code.
+                    cmd.SetGlobalColor(k_LightColorID, light.intensity * light.color);
+
                     if (light.lightType == Light2D.LightType.Sprite && light.lightCookieSprite != null && light.lightCookieSprite.texture != null)
                         cmd.SetGlobalTexture(k_CookieTexID, light.lightCookieSprite.texture);
 
-                    cmd.SetGlobalFloat(k_FalloffIntensityID, light.falloffIntensity);
-                    cmd.SetGlobalColor(k_LightColorID, light.intensity * light.color);
-                    cmd.SetGlobalFloat(k_VolumeOpacityID, light.volumeOpacity);
-
                     if (light.useNormalMap || light.lightType == Light2D.LightType.Point)
                         SetPointLightShaderGlobals(pass, cmd, light);
 
-                    // Light code could be combined...
-                    if (light.lightType == Light2D.LightType.Parametric || light.lightType == Light2D.LightType.Freeform || light.lightType == Light2D.LightType.Sprite)
+                    // Batch if possible.
+                    bool skipDraw = false;
+                    if (batchable)
                     {
-                        cmd.DrawMesh(lightMesh, light.transform.localToWorldMatrix, lightMaterial);
+                        if (light.lightType == Light2D.LightType.Parametric ||
+                            light.lightType == Light2D.LightType.Freeform)
+                        {
+                            if (light.shadowIntensity == 0)
+                            {
+                                if (Light2DBatch.sActiveMaterial == null || lightMaterial != Light2DBatch.sActiveMaterial)
```

Comment:
just because it is the same material, doesn't mean it can batch right? if you try to batch 2 lights that are multiplicative, would it work? Only additive i think...

Reply:
BlendStyles are handled from the caller function and only Additive style is allowed for Batching